Modix.Data/Repositories/PromotionDialogRepository.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Modix.Data.Models.Promotions;
+
+namespace Modix.Data.Repositories
+{
+    public interface IPromotionDialogRepository
+    {
+        /// <summary>
+        /// Creates a new dialog entity and adds it to the database
+        /// </summary>
+        /// <param name="data"></param>
+        /// <returns></returns>
+        Task CreateAsync(ulong messageId, long campaignId);
+
+        /// <summary>
+        /// gets the current active dialogs
+        /// </summary>
+        /// <returns></returns>
+        Task<IReadOnlyList<PromotionDialogBrief>> GetDialogsAsync();
+
+        /// <summary>
+        /// Attempts to delete a poll by its message id
+        /// </summary>
+        /// <param name="messageId"></param>
+        /// <returns></returns>
+        Task<bool> TryDeleteAsync(ulong messageId);
+    }
+
+    public class PromotionDialogRepository : RepositoryBase, IPromotionDialogRepository
+    {
+        public PromotionDialogRepository(ModixContext modixContext) : base(modixContext) { }
+
+        public async Task CreateAsync(ulong messageId, long campaignId)
+        {
+            await ModixContext.Set<PromotionDialogEntity>().AddAsync(new PromotionDialogEntity
+            {
+                MessageId = messageId,
+                CampaignId = campaignId
+            });
+            await ModixContext.SaveChangesAsync();
+        }
+
+        public async Task<IReadOnlyList<PromotionDialogBrief>> GetDialogsAsync()
+            => await ModixContext.Set<PromotionDialogEntity>().AsNoTracking()
+                .Select(PromotionDialogBrief.FromEntityExpression)
```

Comment:
Do we need to call `.AsExpandable()` first?

Reply:
I seem to recall a discussion with @JakenVeina about not needing that, but i could very very well be mistaken or misremembering so im not sure

Reply:
This one in particular doesn't, since there's no nesting projections, right? I mean, it would error out immediately if it was needed.